File: RangeBarSample/src/main/java/com/example/rangebarsample/MainActivity.java

package com.example.rangebarsample;

import android.graphics.Typeface;
import android.os.Bundle;
import android.view.Menu;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;
import android.widget.Button;
import android.widget.EditText;
import android.widget.SeekBar;
import android.widget.SeekBar.OnSeekBarChangeListener;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.edmodo.rangebar.RangeBar;

public class MainActivity extends AppCompatActivity {

    // Sets the initial values such that the image will be drawn
    private static final int DEFAULT_THUMB_COLOR_NORMAL = -1;
    private static final int DEFAULT_THUMB_COLOR_PRESSED = -1;

    // Initializes the RangeBar in the application
    private RangeBar rangebar;

    // Restores the state upon rotating the screen/restarting the activity
    @Override
    protected void onRestoreInstanceState(Bundle bundle) {
        super.onRestoreInstanceState(bundle);

        // Gets the RangeBar
        rangebar = (RangeBar) findViewById(R.id.rangebar1);

        // Gets the index value TextViews
        final TextView leftIndexValue = (TextView) findViewById(R.id.leftIndexValue);
        final TextView rightIndexValue = (TextView) findViewById(R.id.rightIndexValue);
        // Resets the index values every time the activity is changed
        leftIndexValue.setText("" + rangebar.getLeftIndex());
        rightIndexValue.setText("" + rangebar.getRightIndex());

        // Sets focus to the main layout, not the index text fields
        findViewById(R.id.mylayout).requestFocus();

    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Removes title bar and sets content view
        this.requestWindowFeature(Window.FEATURE_NO_TITLE);
        setContentView(R.layout.activity_main);

        // Sets fonts for all
        ViewGroup root = (ViewGroup) findViewById(R.id.mylayout);

        // Gets the buttons references for the buttons
        final Button barColor = (Button) findViewById(R.id.barColor);
        final Button connectingLineColor = (Button) findViewById(R.id.connectingLineColor);
        final Button refreshButton = (Button) findViewById(R.id.refresh);
        
        // Gets the RangeBar
        rangebar = (RangeBar) findViewById(R.id.rangebar1);

        // Setting Index Values -------------------------------

        // Gets the index value TextViews
        final EditText leftIndexValue = (EditText) findViewById(R.id.leftIndexValue);
        final EditText rightIndexValue = (EditText) findViewById(R.id.rightIndexValue);

        // Sets the display values of the indices
        rangebar.setOnRangeBarChangeListener(new RangeBar.OnRangeBarChangeListener() {
            @Override
            public void onIndexChangeListener(RangeBar rangeBar, int leftThumbIndex, int rightThumbIndex) {

                leftIndexValue.setText("" + leftThumbIndex);
                rightIndexValue.setText("" + rightThumbIndex);
            }
        });

        // Sets the indices themselves upon input from the user
        refreshButton.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {

                // Gets the String values of all the texts
                String leftIndex = leftIndexValue.getText().toString();
                String rightIndex = rightIndexValue.getText().toString();

                // Catches any IllegalArgumentExceptions; if fails, should throw
                // a dialog warning the user
                try {
                    if (!leftIndex.isEmpty() && !rightIndex.isEmpty())
                    {
                        int leftIntIndex = Integer.parseInt(leftIndex);
                        int rightIntIndex = Integer.parseInt(rightIndex);
                        rangebar.setThumbIndices(leftIntIndex, rightIntIndex);
                    }
                } catch (IllegalArgumentException e) {
                }
            }
        });

        // Setting Number Attributes -------------------------------

        // Sets tickCount
        final TextView tickCount = (TextView) findViewById(R.id.tickCount);
        SeekBar tickCountSeek = (SeekBar) findViewById(R.id.tickCountSeek);
        tickCountSeek.setOnSeekBarChangeListener(new OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar tickCountSeek, int progress, boolean fromUser) {
                try {
                    rangebar.setTickCount(progress);
                } catch (IllegalArgumentException e) {
                }
                tickCount.setText("tickCount = " + progress);
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {
            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {
            }
        });

        // Sets tickHeight
        final TextView tickHeight = (TextView) findViewById(R.id.tickHeight);
        SeekBar tickHeightSeek = (SeekBar) findViewById(R.id.tickHeightSeek);
        tickHeightSeek.setOnSeekBarChangeListener(new OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar tickHeightSeek, int progress, boolean fromUser) {
                rangebar.setTickHeight(progress);
                tickHeight.setText("tickHeight = " + progress);
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {
            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {
            }
        });

        // Sets barWeight
        final TextView barWeight = (TextView) findViewById(R.id.barWeight);
        SeekBar barWeightSeek = (SeekBar) findViewById(R.id.barWeightSeek);
        barWeightSeek.setOnSeekBarChangeListener(new OnSeekBarChangeListener() {

            @Override
            public void onProgressChanged(SeekBar barWeightSeek, int progress, boolean fromUser) {
                rangebar.setBarWeight(progress);
                barWeight.setText("barWeight = " + progress);
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {
            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {
            }
        });

        // Sets connectingLineWeight
        final TextView connectingLineWeight = (TextView) findViewById(R.id.connectingLineWeight);
        SeekBar connectingLineWeightSeek = (SeekBar) findViewById(R.id.connectingLineWeightSeek);
        connectingLineWeightSeek.setOnSeekBarChangeListener(new OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar connectingLineWeightSeek, int progress, boolean fromUser) {
                rangebar.setConnectingLineWeight(progress);
                connectingLineWeight.setText("connectingLineWeight = " + progress);
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {
            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {
            }
        });

        // Sets thumbRadius
        final TextView thumbRadius = (TextView) findViewById(R.id.thumbRadius);
        SeekBar thumbRadiusSeek = (SeekBar) findViewById(R.id.thumbRadiusSeek);
        thumbRadiusSeek.setOnSeekBarChangeListener(new OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar thumbRadiusSeek, int progress, boolean fromUser) {
                if (progress == 0) {
                    rangebar.setThumbRadius(-1);
                    thumbRadius.setText("thumbRadius = N/A");
                }
                else {
                    rangebar.setThumbRadius(progress);
                    thumbRadius.setText("thumbRadius = " + progress);
                }
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {
            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {
            }
        });

        // Setting Color Attributes---------------------------------



    }



    /**
     * Sets the font on all TextViews in the ViewGroup. Searches recursively for
     * all inner ViewGroups as well. Just add a check for any other views you
     * want to set as well (EditText, etc.)
     */
    private void setFont(ViewGroup group, Typeface font) {
        int count = group.getChildCount();
        View v;
        for (int i = 0; i < count; i++) {
            v = group.getChildAt(i);
            if (v instanceof TextView || v instanceof EditText || v instanceof Button) {
                ((TextView) v).setTypeface(font);
            } else if (v instanceof ViewGroup)
                setFont((ViewGroup) v, font);
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.main, menu);
        return true;
    }

}

File: RangeBarSample/src/main/java/com/example/rangebarsample/MainActivity.kt
package com.example.rangebarsample

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.Button
import android.widget.SeekBar
import android.widget.SeekBar.OnSeekBarChangeListener
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.edmodo.rangebar.RangeBar

class MainActivity : AppCompatActivity() {
    // Initializes the RangeBar in the application
    private val rangebar by lazy { findViewById<RangeBar>(R.id.rangebar1) }
    private val refreshButton by lazy { findViewById<Button>(R.id.refresh) }
    private val thumbRadius by lazy { findViewById<TextView>(R.id.thumbRadius) }
    private val tickCount by lazy { findViewById<TextView>(R.id.tickCount) }
    private val tickCountSeek by lazy { findViewById<SeekBar>(R.id.tickCountSeek) }
    private val leftIndexValue by lazy { findViewById<TextView>(R.id.leftIndexValue) }
    private val tickHeight by lazy { findViewById<TextView>(R.id.tickHeight) }
    private val rightIndexValue by lazy { findViewById<TextView>(R.id.rightIndexValue) }
    private val thumbRadiusSeek by lazy { findViewById<SeekBar>(R.id.thumbRadiusSeek) }
    private val tickHeightSeek by lazy { findViewById<SeekBar>(R.id.tickHeightSeek) }
    private val barWeight by lazy { findViewById<TextView>(R.id.barWeight) }
    private val connectingLineWeight by lazy { findViewById<TextView>(R.id.connectingLineWeight) }
    private val barWeightSeek by lazy { findViewById<SeekBar>(R.id.barWeightSeek) }
    private val connectingLineWeightSeek by lazy { findViewById<SeekBar>(R.id.connectingLineWeightSeek) }

    override fun onRestoreInstanceState(bundle: Bundle) {
        super.onRestoreInstanceState(bundle)
        // Resets the index values every time the activity is changed
        leftIndexValue.text = "${rangebar.leftIndex}"
        rightIndexValue.text = "${rangebar!!.rightIndex}"
        // Sets focus to the main layout, not the index text fields
        findViewById<View>(R.id.mylayout).requestFocus()
    }


    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == R.id.layout_activity) {
            startActivity(Intent(this, LayoutPreviewActivity::class.java))
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        rangebar.setOnRangeBarChangeListener(object : RangeBar.OnRangeBarChangeListener {
            override fun onIndexChangeListener(
                rangeBar: RangeBar?,
                leftThumbIndex: Int,
                rightThumbIndex: Int
            ) {
                leftIndexValue.text = "$leftThumbIndex"
                rightIndexValue.text = "$rightThumbIndex"
            }
        })

        // Sets the indices themselves upon input from the user
        refreshButton.setOnClickListener { // Gets the String values of all the texts
            val leftIndex = leftIndexValue.text.toString()
            val rightIndex = rightIndexValue.text.toString()

            try {
                if (leftIndex.isNotEmpty() && rightIndex.isNotEmpty()) {
                    val leftIntIndex = leftIndex.toInt()
                    val rightIntIndex = rightIndex.toInt()
                    rangebar!!.setThumbIndices(leftIntIndex, rightIntIndex)
                }
            } catch (e: IllegalArgumentException) {
            }
        }

        tickCountSeek.setOnSeekBarChangeListener(object : OnSeekBarChangeListener {
            override fun onProgressChanged(
                tickCountSeek: SeekBar,
                progress: Int,
                fromUser: Boolean
            ) {
                try {
                    rangebar!!.setTickCount(progress)
                } catch (e: IllegalArgumentException) {
                }
                tickCount.text = "tickCount = $progress"
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {}
            override fun onStopTrackingTouch(seekBar: SeekBar) {}
        })

        tickHeightSeek.setOnSeekBarChangeListener(object : OnSeekBarChangeListener {
            override fun onProgressChanged(
                tickHeightSeek: SeekBar,
                progress: Int,
                fromUser: Boolean
            ) {
                rangebar!!.setTickHeight(progress.toFloat())
                tickHeight.text = "tickHeight = $progress"
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {}
            override fun onStopTrackingTouch(seekBar: SeekBar) {}
        })

        barWeightSeek.setOnSeekBarChangeListener(object : OnSeekBarChangeListener {
            override fun onProgressChanged(
                barWeightSeek: SeekBar,
                progress: Int,
                fromUser: Boolean
            ) {
                rangebar!!.setBarWeight(progress.toFloat())
                barWeight.text = "barWeight = $progress"
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {}
            override fun onStopTrackingTouch(seekBar: SeekBar) {}
        })

        connectingLineWeightSeek.setOnSeekBarChangeListener(object : OnSeekBarChangeListener {
            override fun onProgressChanged(
                connectingLineWeightSeek: SeekBar,
                progress: Int,
                fromUser: Boolean
            ) {
                rangebar!!.setConnectingLineWeight(progress.toFloat())
                connectingLineWeight.text = "connectingLineWeight = $progress"
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {}
            override fun onStopTrackingTouch(seekBar: SeekBar) {}
        })
        thumbRadiusSeek.setOnSeekBarChangeListener(object : OnSeekBarChangeListener {
            override fun onProgressChanged(
                thumbRadiusSeek: SeekBar,
                progress: Int,
                fromUser: Boolean
            ) {
                if (progress == 0) {
                    rangebar!!.setThumbRadius(-1f)
                    thumbRadius.text = "thumbRadius = N/A"
                } else {
                    rangebar!!.setThumbRadius(progress.toFloat())
                    thumbRadius.text = "thumbRadius = $progress"
                }
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {}
            override fun onStopTrackingTouch(seekBar: SeekBar) {}
        })
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.main, menu)
        return true
    }

}
